plugins/out_s3/s3.c

```diff
@@ -981,11 +1018,23 @@ static int upload_data(struct flb_s3 *ctx, struct s3_file *chunk,
     int size_check = FLB_FALSE;
     int part_num_check = FLB_FALSE;
     int timeout_check = FLB_FALSE;
-    time_t create_time;
```

Comment:
comment above is wrong:

	 * return value is one of 0, -1, -2.	
 * 0 means uploaded successfully.	
 * -1 means failed to upload data, will retry.	
 * -2 means failed to upload data, but can't retry because reach the retry_limit.

The may13th one commit branch has most up to date changes that need to be sync'd here.

Reply:
Return Value is wrong here too. Another issue from rebase.